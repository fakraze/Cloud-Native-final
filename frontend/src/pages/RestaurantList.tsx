import React from 'react';
import { Link } from 'react-router-dom';
import { Star, Clock, MapPin } from 'lucide-react';
import { useRestaurants } from '../hooks/useRestaurant';

const RestaurantList: React.FC = () => {
  const { data: restaurants, isLoading, error } = useRestaurants();

  if (isLoading) {
    return (
      <div className="flex items-center justify-center h-64">
        <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-blue-600"></div>
      </div>
    );
  }

  if (error) {
    return (
      <div className="bg-red-50 border border-red-200 rounded-lg p-4">
        <p className="text-red-600">Failed to load restaurants. Please try again.</p>
      </div>
    );
  }

  return (
    <div className="max-w-7xl mx-auto">
      <div className="mb-8">
        <h1 className="text-3xl font-bold text-gray-900">Restaurants</h1>
        <p className="text-gray-600 mt-2">Choose from our partner restaurants</p>
      </div>

      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6">
        {restaurants?.map((restaurant) => (
          <Link
            key={restaurant.id}
            to={`/restaurant/${restaurant.id}`}
            className="card hover:shadow-lg transition-shadow duration-200 overflow-hidden"
          >
            <div className="aspect-w-16 aspect-h-9">
              <img
                src={restaurant.imageUrl || '/placeholder-restaurant.jpg'}
                alt={restaurant.name}
                className="w-full h-48 object-cover"
              />
            </div>
            
            <div className="p-6">
              <div className="flex items-start justify-between mb-3">
                <h3 className="text-xl font-semibold text-gray-900 line-clamp-1">
                  {restaurant.name}
                </h3>
                <div className="flex items-center space-x-1">
                  <Star className="h-4 w-4 text-yellow-400 fill-current" />
                  <span className="text-sm text-gray-600">{restaurant.rating.toFixed(1)}</span>
                </div>
              </div>
              
              <p className="text-gray-600 text-sm mb-4 line-clamp-2">
                {restaurant.description}
              </p>
              
              <div className="flex items-center justify-between text-sm text-gray-500">
                <div className="flex items-center space-x-1">
                  <Clock className="h-4 w-4" />
                  <span>15-30 min</span>
                </div>
                <div className="flex items-center space-x-1">
                  <MapPin className="h-4 w-4" />
                  <span>Pickup available</span>
                </div>
              </div>
              
              {!restaurant.isActive && (
                <div className="mt-3 bg-red-50 border border-red-200 rounded px-3 py-1">
                  <span className="text-xs text-red-600 font-medium">Currently Closed</span>
                </div>
              )}
            </div>
          </Link>
        ))}
      </div>

      {restaurants?.length === 0 && (
        <div className="text-center py-12">
          <p className="text-gray-500 text-lg">No restaurants available at the moment.</p>
        </div>
      )}
    </div>
  );
};

export { RestaurantList };
