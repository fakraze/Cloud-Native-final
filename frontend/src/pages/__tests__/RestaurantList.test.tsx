import { describe, it, expect, beforeEach, vi } from 'vitest';
import { render } from '../../test/test-utils';
import { RestaurantList } from '../RestaurantList';

// Mock the hooks
vi.mock('../../hooks/useRestaurant', () => ({
  useRestaurants: vi.fn()
}));

const mockUseRestaurants = vi.hoisted(() => ({
  useRestaurants: vi.fn()
}));

vi.mock('../../hooks/useRestaurant', () => mockUseRestaurants);

describe('RestaurantList Component', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('should render loading state', () => {
    mockUseRestaurants.useRestaurants.mockReturnValue({
      data: null,
      isLoading: true,
      error: null
    });

    const { container } = render(<RestaurantList />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render error state', () => {
    mockUseRestaurants.useRestaurants.mockReturnValue({
      data: null,
      isLoading: false,
      error: new Error('Failed to fetch')
    });

    const { container } = render(<RestaurantList />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render restaurant list with data', () => {
    mockUseRestaurants.useRestaurants.mockReturnValue({
      data: [
        {
          id: '1',
          name: 'Italiano Restaurant',
          description: 'Authentic Italian cuisine',
          cuisine: 'Italian',
          rating: 4.5,
          imageUrl: '/images/italian.jpg',
          address: '123 Main St',
          deliveryTime: '30-40 min'
        },
        {
          id: '2',
          name: 'Burger House',
          description: 'Best burgers in town',
          cuisine: 'American',
          rating: 4.2,
          imageUrl: '/images/burger.jpg',
          address: '456 Oak Ave',
          deliveryTime: '20-30 min'
        },
        {
          id: '3',
          name: 'Sushi Palace',
          description: 'Fresh sushi and Japanese dishes',
          cuisine: 'Japanese',
          rating: 4.8,
          imageUrl: '/images/sushi.jpg',
          address: '789 Pine St',
          deliveryTime: '25-35 min'
        }
      ],
      isLoading: false,
      error: null
    });

    const { container } = render(<RestaurantList />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render empty restaurant list', () => {
    mockUseRestaurants.useRestaurants.mockReturnValue({
      data: [],
      isLoading: false,
      error: null
    });

    const { container } = render(<RestaurantList />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
