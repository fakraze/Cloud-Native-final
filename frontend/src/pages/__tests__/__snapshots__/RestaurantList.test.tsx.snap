// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RestaurantList Component > should render empty restaurant list 1`] = `
<div
  class="max-w-7xl mx-auto"
>
  <div
    class="mb-8"
  >
    <h1
      class="text-3xl font-bold text-gray-900"
    >
      Restaurants
    </h1>
    <p
      class="text-gray-600 mt-2"
    >
      Choose from our partner restaurants
    </p>
  </div>
  <div
    class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6"
  />
  <div
    class="text-center py-12"
  >
    <p
      class="text-gray-500 text-lg"
    >
      No restaurants available at the moment.
    </p>
  </div>
</div>
`;

exports[`RestaurantList Component > should render error state 1`] = `
<div
  class="bg-red-50 border border-red-200 rounded-lg p-4"
>
  <p
    class="text-red-600"
  >
    Failed to load restaurants. Please try again.
  </p>
</div>
`;

exports[`RestaurantList Component > should render loading state 1`] = `
<div
  class="flex items-center justify-center h-64"
>
  <div
    class="animate-spin rounded-full h-8 w-8 border-b-2 border-blue-600"
  />
</div>
`;

exports[`RestaurantList Component > should render restaurant list with data 1`] = `
<div
  class="max-w-7xl mx-auto"
>
  <div
    class="mb-8"
  >
    <h1
      class="text-3xl font-bold text-gray-900"
    >
      Restaurants
    </h1>
    <p
      class="text-gray-600 mt-2"
    >
      Choose from our partner restaurants
    </p>
  </div>
  <div
    class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6"
  >
    <a
      class="card hover:shadow-lg transition-shadow duration-200 overflow-hidden"
      data-discover="true"
      href="/restaurant/1"
    >
      <div
        class="aspect-w-16 aspect-h-9"
      >
        <img
          alt="Italiano Restaurant"
          class="w-full h-48 object-cover"
          src="/images/italian.jpg"
        />
      </div>
      <div
        class="p-6"
      >
        <div
          class="flex items-start justify-between mb-3"
        >
          <h3
            class="text-xl font-semibold text-gray-900 line-clamp-1"
          >
            Italiano Restaurant
          </h3>
          <div
            class="flex items-center space-x-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-4 w-4 text-yellow-400 fill-current"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <span
              class="text-sm text-gray-600"
            >
              4.5
            </span>
          </div>
        </div>
        <p
          class="text-gray-600 text-sm mb-4 line-clamp-2"
        >
          Authentic Italian cuisine
        </p>
        <div
          class="flex items-center justify-between text-sm text-gray-500"
        >
          <div
            class="flex items-center space-x-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-clock h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <polyline
                points="12 6 12 12 16 14"
              />
            </svg>
            <span>
              15-30 min
            </span>
          </div>
          <div
            class="flex items-center space-x-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-map-pin h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
              />
              <circle
                cx="12"
                cy="10"
                r="3"
              />
            </svg>
            <span>
              Pickup available
            </span>
          </div>
        </div>
        <div
          class="mt-3 bg-red-50 border border-red-200 rounded px-3 py-1"
        >
          <span
            class="text-xs text-red-600 font-medium"
          >
            Currently Closed
          </span>
        </div>
      </div>
    </a>
    <a
      class="card hover:shadow-lg transition-shadow duration-200 overflow-hidden"
      data-discover="true"
      href="/restaurant/2"
    >
      <div
        class="aspect-w-16 aspect-h-9"
      >
        <img
          alt="Burger House"
          class="w-full h-48 object-cover"
          src="/images/burger.jpg"
        />
      </div>
      <div
        class="p-6"
      >
        <div
          class="flex items-start justify-between mb-3"
        >
          <h3
            class="text-xl font-semibold text-gray-900 line-clamp-1"
          >
            Burger House
          </h3>
          <div
            class="flex items-center space-x-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-4 w-4 text-yellow-400 fill-current"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <span
              class="text-sm text-gray-600"
            >
              4.2
            </span>
          </div>
        </div>
        <p
          class="text-gray-600 text-sm mb-4 line-clamp-2"
        >
          Best burgers in town
        </p>
        <div
          class="flex items-center justify-between text-sm text-gray-500"
        >
          <div
            class="flex items-center space-x-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-clock h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <polyline
                points="12 6 12 12 16 14"
              />
            </svg>
            <span>
              15-30 min
            </span>
          </div>
          <div
            class="flex items-center space-x-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-map-pin h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
              />
              <circle
                cx="12"
                cy="10"
                r="3"
              />
            </svg>
            <span>
              Pickup available
            </span>
          </div>
        </div>
        <div
          class="mt-3 bg-red-50 border border-red-200 rounded px-3 py-1"
        >
          <span
            class="text-xs text-red-600 font-medium"
          >
            Currently Closed
          </span>
        </div>
      </div>
    </a>
    <a
      class="card hover:shadow-lg transition-shadow duration-200 overflow-hidden"
      data-discover="true"
      href="/restaurant/3"
    >
      <div
        class="aspect-w-16 aspect-h-9"
      >
        <img
          alt="Sushi Palace"
          class="w-full h-48 object-cover"
          src="/images/sushi.jpg"
        />
      </div>
      <div
        class="p-6"
      >
        <div
          class="flex items-start justify-between mb-3"
        >
          <h3
            class="text-xl font-semibold text-gray-900 line-clamp-1"
          >
            Sushi Palace
          </h3>
          <div
            class="flex items-center space-x-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-4 w-4 text-yellow-400 fill-current"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <span
              class="text-sm text-gray-600"
            >
              4.8
            </span>
          </div>
        </div>
        <p
          class="text-gray-600 text-sm mb-4 line-clamp-2"
        >
          Fresh sushi and Japanese dishes
        </p>
        <div
          class="flex items-center justify-between text-sm text-gray-500"
        >
          <div
            class="flex items-center space-x-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-clock h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <polyline
                points="12 6 12 12 16 14"
              />
            </svg>
            <span>
              15-30 min
            </span>
          </div>
          <div
            class="flex items-center space-x-1"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-map-pin h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
              />
              <circle
                cx="12"
                cy="10"
                r="3"
              />
            </svg>
            <span>
              Pickup available
            </span>
          </div>
        </div>
        <div
          class="mt-3 bg-red-50 border border-red-200 rounded px-3 py-1"
        >
          <span
            class="text-xs text-red-600 font-medium"
          >
            Currently Closed
          </span>
        </div>
      </div>
    </a>
  </div>
</div>
`;
